flink-clients/src/test/java/org/apache/flink/client/deployment/application/ApplicationDispatcherBootstrapTest.java

```diff
@@ -348,6 +351,11 @@ void testApplicationIsStoppedWhenStoppingBootstrap() throws Exception {
 
         bootstrap.stop();
 
+        // Triggers the scheduled ApplicationDispatcherBootstrap process after calling stop. This
+        // ensures that the bootstrap task isn't completed before the stop method is called which
+        // would prevent the stop call from cancelling the task's future.
+        manuallyTriggeredExecutor.triggerNonPeriodicScheduledTask();
```

Comment:
hmm. I guess this is fine, but my impression is that the manual executor relies more on implementation details than delaying the job submission would.